dotnet/src/SemanticKernel/Orchestration/SKFunctionExtensions.cs

```diff
@@ -59,7 +51,9 @@ public static ISKFunction UseTemperature(this ISKFunction skFunction, double tem
     /// <returns>Self instance</returns>
     public static ISKFunction UseTopP(this ISKFunction skFunction, double topP)
     {
-        skFunction.RequestSettings.TopP = topP;
+        Verify.NotNull(skFunction.AIRequestSettings.CompleteRequestSettings, "Completion request settings cannot be empty");
+        Verify.WithinRangeInclusive<double>(topP, 0.0, 1.0, "topP");
```

Comment:
this and other ranges below are model specific. I'm quite concerned about a number of changes in the PR that assume that the kernel works only with HTTP and only with OpenAI